devicons.py

```diff
@@ -8,22 +8,16 @@
 
 # Get the XDG_USER_DIRS directory names from enviromental variables
 
-XDG_DOCUMENTS_DIR = os.getenv('XDG_DOCUMENTS_DIR', default=' ')
-XDG_DOCUMENTS_DIR = XDG_DOCUMENTS_DIR.split('/')[-2]
-XDG_DOWNLOAD_DIR = os.getenv('XDG_DOWNLOAD_DIR', default=' ')
-XDG_DOWNLOAD_DIR = XDG_DOWNLOAD_DIR.split('/')[-2]
-XDG_CONFIG_DIR = os.getenv('XDG_CONFIG_DIR', default=' ')
-XDG_CONFIG_DIR = XDG_CONFIG_DIR.split('/')[-2]
-XDG_MUSIC_DIR = os.getenv('XDG_MUSIC_DIR', default=' ')
-XDG_MUSIC_DIR = XDG_MUSIC_DIR.split('/')[-2]
-XDG_PICTURES_DIR = os.getenv('XDG_PICTURES_DIR', default=' ')
-XDG_PICTURES_DIR = XDG_PICTURES_DIR.split('/')[-2]
-XDG_PUBLICSHARE_DIR = os.getenv('XDG_PUBLICSHARE_DIR', default=' ')
-XDG_PUBLICSHARE_DIR = XDG_PUBLICSHARE_DIR.split('/')[-2]
-XDG_TEMPLATES_DIR = os.getenv('XDG_TEMPLATES_DIR', default=' ')
-XDG_TEMPLATES_DIR = XDG_TEMPLATES_DIR.split('/')[-2]
-XDG_VIDEOS_DIR = os.getenv('XDG_VIDEOS_DIR', default=' ')
-XDG_VIDEOS_DIR = XDG_VIDEOS_DIR.split('/')[-2]
+xdgs_dirs = {path.split('/')[-2]: icon for key, icon in [
+    ('XDG_DOCUMENTS_DIR'  , ''),
+    ('XDG_DOWNLOAD_DIR'   , ''),
+    ('XDG_CONFIG_DIR'     , ''),
+    ('XDG_MUSIC_DIR'      , ''),
+    ('XDG_PICTURES_DIR'   , ''),
+    ('XDG_PUBLICSHARE_DIR', ''),
+    ('XDG_TEMPLATES_DIR'  , ''),
+    ('XDG_VIDEOS_DIR'     , ''),
+] if (path := os.getenv(key))}
```

Comment:
Theoretically, Ranger supports Python all from 2.6, 2.7, to 3.1~3.10. But new syntax `:=` only available on `Python>=3.8`.

https://github.com/ranger/ranger/blob/1de6787c1c8bf5d33ad450bc81b1a0495da71422/setup.py#L79-L87

On some Linux distributions, the default Python3 version has not been updated yet (e.g. on Ubuntu 18.04 it's Python 3.6).